orttraining/orttraining/training_api/include/module.h

```diff
@@ -29,15 +30,16 @@ struct Parameter {
   std::string GradientName() const { return gradient_name_; }
 
   // Reset and release the gradient buffer of this Parameter.
-  Status ResetGrad() {
-    return Status::OK();
-  }
+  Status ResetGrad();
 
   Status SetRequiresGrad(bool requires_grad) {
     requires_grad_ = requires_grad;
     return Status::OK();
```

Comment:
Do we want to allow toggling requires_grad attr?  If not it is best we add this to constructor and remove this mtd.